README.md

```diff
@@ -0,0 +1,564 @@
+
+AuthorizationRequest.Builder authorizationRequest = new AuthorizationRequest.Builder(
+        serviceConfiguration,
+  mobileService.getKClientId(),
+  ResponseTypeValues.CODE,
+  Uri.parse(k_REDIRECT_URI));
+
+AuthorizationRequest authRequest = authorizationRequest.build();
+
+Intent authIntent = mAuthService.getAuthorizationRequestIntent(authRequest);
+startActivityForResult(authIntent, RC_AUTH);
+```
+- It is important to note that we need to provide valid `redirect_URI` and `clientId`, in our sample app we are using `mobile-services.json` file that has been generated in [MDC]([https://access.redhat.com/documentation/en-us/red_hat_mobile_developer_services/1/pdf/getting_started/Red_Hat_Mobile_Developer_Services-1-Getting_Started-en-US.pdf](https://access.redhat.com/documentation/en-us/red_hat_mobile_developer_services/1/pdf/getting_started/Red_Hat_Mobile_Developer_Services-1-Getting_Started-en-US.pdf)). Once authorization request has been send app is redirected to Keycloak log in screen.
+- After logging in with valid credentials and providing valid `redirect_URI` user is pushed back to the application and can retrieve response and can update state in auth state manager if wish to do so.
+```java
+@Override
+protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+ if (requestCode == RC_AUTH) {
+        AuthorizationResponse resp = AuthorizationResponse.fromIntent(data);
+  AuthorizationException ex = AuthorizationException.fromIntent(data);
+
+  OPTIONALLY UPDATE AUTH STATE MANAGER
+
+  exchangeAuthorizationCode(resp);
+
+  } else {
+        System.out.println("Error, wrong response code");
+  }
+}
+```
+- With the response back we can trigger `exchangeAuthorizationCode` and perform token request
+```java
+private void exchangeAuthorizationCode(AuthorizationResponse authorizationResponse) {
+    performTokenRequest(authorizationResponse.createTokenExchangeRequest());
+}
+private void performTokenRequest(TokenRequest request) {
+    mAuthService.performTokenRequest(
+            request,
+  (tokenResponse, ex) -> {
+                receivedTokenResponse(tokenResponse, ex);
+  mAuthStateManager.updateAfterTokenResponse(tokenResponse, ex);
+  });
+}
+```
+- Once we get a response we can view our token details
+```java
+private void receivedTokenResponse(
+        @Nullable TokenResponse tokenResponse,
+  @Nullable AuthorizationException authException) {
+    mAuthState.update(tokenResponse, authException);
+
+  System.out.println("Token expires in : " + mAuthState.getAccessTokenExpirationTime());
+
+  startActivity(new Intent(this, MainActivity.class));
+}
+```
+### 3. Unifiedpush implementation
+#### External Setup
+For setting up push notifications a [firebase][https://firebase.google.com/] account is required. The following information below is required and can be found the firebase project settings.
```

Comment:
@Boomatang 
does this syntax works with md? shouldn't it be [ name ]( address ) ?